src/consumers.ts

```diff
@@ -34,7 +36,7 @@ interface IConsumer {
 
 /* Generic `T` producer */
 interface IProducer<T> {
-    produce(ctx: any): T;
+    produce(ctx: any, cls: DiagnosticsPipeline): T;
```

Comment:
This is an anti pattern. Why should a abstract generic producer deals with concrete DiagnosticsPipeline? Can't we dedup the old Diagnostics after producing it?

e.g consider you have response for 3packages under single module m/a, m/b, m/c
In this case you will have {D1, D1 + D2, D1 + D2 + D3}. If you dedup the Diagnostic after the pipeline, you will have only { D1+ D2 + D3} 

Note: + denotes aggregation

Reply:
This is modified now.